### PR TITLE
Add sub

### PR DESCRIPTION
**Pull request by chongr**

### CODE REVIEW

**Comment by chongr** [comment]
hmm it looks like the subscriptions are prefixed with `hedwig-firehose` right now should we replace that firehose string too?

File: gcp/gcp.go [hunk 1]
```diff
@@ -109,9 +114,10 @@ func (b *Backend) Receive(ctx context.Context, numMessages uint32, visibilityTim
 					message.DeliveryAttempt = &deliveryAttemptDefault
 				}
 				metadata := Metadata{
-					pubsubMessage:   message,
-					PublishTime:     message.PublishTime,
-					DeliveryAttempt: *message.DeliveryAttempt,
+					pubsubMessage:    message,
+					PublishTime:      message.PublishTime,
+					DeliveryAttempt:  *message.DeliveryAttempt,
+					SubscriptionName: strings.Replace(subID, "hedwig", "", 1),
```
Comment (maroux) [inline gcp/gcp.go:120]: Huh shouldn't it be "hedwig-". And maybe to be defensive it should be remove prefix not replace all.

**Review by maroux — CHANGES_REQUESTED** [review]

File: gcp/gcp.go (hunk 1 above)
Comment (chongr) [reply gcp/gcp.go]: ah yeah I caught that in the test and just changed - but will change to only remove at the beginning

**Comment by maroux** [comment]
> hmm it looks like the subscriptions are prefixed with `hedwig-firehose` right now should we replace that firehose string too?

Yeah we can replace firehose to calculate directory name but that should be in firehose code base not here.

**Review by maroux — APPROVED** [review]

**Pull request merged**